### ISSUE TITLE
Can tispark connect to multiple tidb clusters in the same time ?

### ISSUE THREAD

**Issue by louisliu318** [pingcap/tispark#554]
Can tispark connect to multiple tidb clusters in the same time ?

**Comment by birdstorm** [comment]
No, TiSpark does not support multiple TiDB clusters now because it needs to specify PD address upon startup, which of course, differs from each cluster. 

However, if you do not want to join or query data from different clusters, i.e, you wish to support multi-cluster connections rather than queries, it should be possible to implement because we can create respective tikv-client for each cluster. 

We might discuss about your needs and its possible implementation in this issue. It would be nice if you could support with a pull request:) Thanks~

**Comment by louisliu318** [comment]
@birdstorm Thanks. What I want to do is  making tispark as united access entry for all sources of data, i.e , diffrent tidb clusters, hive,mysql.

I didn't get it  about `support multi-cluster connections rather than queries`. Could you explain more clearly.

**Comment by birdstorm** [comment]
@louisliu318 Well, current implementation restricts that TiSpark only accepts PD addresses from one cluster. If you want to join tables from different data sources, say, two TiDB clusters, it will not work by design. However, TiSpark already supports reading and joining data from Hive and TiDB. You may check the documents for details.

If the databases in Hive and TiDB are not identical, you may simply specify the database name in your query before tableIdentifier. In the other hand, you can configure `spark.tispark.db_prefix` to add database prefix before TiDB databases, e.g., `tidb_`, so as to query specific TiDB data by using ``` `tidb_whateverdatabasename`.`whatevertablename` ```
